### PR TITLE
Refactor egardiaserver

### PR DESCRIPTION
**Pull request by MartinHjelmare**
* THIS IS A BREAKING CHANGE.
* Remove home assistant interface.
* Use socketserver module.
* Allow registering callbacks. Callbacks should accept a dict as single argument. The dict will contain the new status of the alarm.
* Add test client module.
* Note that reusing the server address currently doesn't work for me, even though that option is set to True. I need to exit my IPython promt between testing the server.

### CODE REVIEW

File: src/pythonegardia/egardiaserver.py [hunk 1]
```diff
@@ -1,152 +1,115 @@
...
-            _LOGGER.error(msg)
-            sys.exit(2)
-        listen_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        listen_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            self.allow_reuse_address = True
+            self.server_bind()
+            self.server_activate()
+        except OSError:
+            self.server_close()
+            _LOGGER.exception(
```
Comment (MartinHjelmare) [inline src/pythonegardia/egardiaserver.py:33]: This should probably be set to error level before merging this. I changed it to exception level when debugging.

**Review by jeroenterheerdt — APPROVED** [review]

**Comment by jeroenterheerdt** [comment]
Martin - this seems fine. Thanks a lot! How can I now "integrate" this into the alarm_control_panel egardia component? Right now it seems to me that I would still need to run the egardiaserver myself, separately of the HASS install. Not sure if that is correct, though. I was hoping to get the "Egardiaserver" to start automatically when you configure the egardia component in HASS. Would we have to create a gateway for that? How would that work?

**Comment by MartinHjelmare** [comment]
You can instantiate the server during setup of the alarm_control_panel in home assistant, register a callback for events etc. So no need to run anything separately from home assistant anymore. Of course the user still has to change the address to home assistant ip and correct port in the egardia control panel, so I guess this should still be optional.

**Comment by jeroenterheerdt** [comment]
@MartinHjelmare thanks, I have updated the HA component and did the PR. Many thanks again for helping out!

**Pull request merged**